src/Avalonia.Controls/ComboBox.cs

```diff
@@ -291,6 +293,7 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
 
             _popup = e.NameScope.Get<Popup>("PART_Popup");
             _popup.Opened += PopupOpened;
+            _popup.Closed += PopupClosed;
```

Comment:
Huh, wow, looks like we weren't subscribing to the `Closed` event, meaning that `PopupClosed` was never called!